src/meta/src/rpc/ddl_controller.rs

```diff
@@ -866,22 +1060,31 @@ impl DdlController {
         let job_id = stream_job.id();
         tracing::debug!(id = job_id, "creating stream job");
 
-        let result = self
-            .stream_manager
-            .create_streaming_job(table_fragments, ctx)
-            .await;
+        let result = try {
+            // Add table fragments to meta store with state: `State::Initial`.
+            mgr.fragment_manager
+                .start_create_table_fragments(table_fragments.clone())
+                .await?;
+
+            self.stream_manager
+                .create_streaming_job(table_fragments, ctx)
+                .await?
+        };
 
         if let Err(e) = result {
             match stream_job.create_type() {
                 CreateType::Background => {
                     tracing::error!(id = job_id, error = ?e, "finish stream job failed");
-                    if let Err(err) = self
+                    let should_cancel = match mgr
                         .fragment_manager
                         .select_table_fragments_by_table_id(&job_id.into())
                         .await
-                        && err.is_fragment_not_found()
                     {
-                        // If the table fragments are not found, it means that the stream job has not been created.
+                        Err(err) => err.is_fragment_not_found(),
+                        Ok(table_fragments) => table_fragments.is_initial(),
```

Comment:
Fixed a corner case: if the table fragments exists but is still in initial state, it should also should be cancelled. Because the actors are not marked as running yet and their upstream dispatchers are not persist after associated barrier get collected.